railties/lib/rails/dev_caching.rb

```diff
@@ -15,6 +15,7 @@ def enable_by_file
         end
 
         FileUtils.touch 'tmp/restart.txt'
+        FileUtils.rm_f('tmp/pids/server.pid')
```

Comment:
Let's extend the existing test which asserts that the temp file has been removed to also assert that the pid file is removed.